src/android/IntercomIntentService.java

```diff
@@ -0,0 +1,46 @@
+package io.intercom.android.sdk;
+
+import android.app.IntentService;
+import android.content.Context;
+import android.content.Intent;
+import android.content.pm.ResolveInfo;
+
+import java.util.List;
+
+import io.intercom.android.sdk.push.IntercomPushClient;
+
+public class IntercomIntentService extends IntentService {
+    private final IntercomPushClient intercomPushClient = new IntercomPushClient();
+
+    public IntercomIntentService() {
+        super("Intercom Cordova Intent Service");
+    }
+
+    @Override
+    protected void onHandleIntent(Intent intent) {
+        if (intercomPushClient.isIntercomPush(intent.getExtras())) {
+            intercomPushClient.handlePush(getApplication(), intent.getExtras());
+            return;
+        }
+
+        Intent passThroughIntent = new Intent(intent);
+        passThroughIntent.setComponent(null);
+
+        List<ResolveInfo> services = getPackageManager().queryIntentServices(passThroughIntent, 0);
+        for (ResolveInfo info : services) {
+            try {
+                Class serviceClass = Class.forName(info.serviceInfo.name);
+                if (serviceClass == this.getClass()) {
+                    continue;
+                }
+
+                Context applicationContext = getApplicationContext();
+                passThroughIntent.setClass(applicationContext, serviceClass);
+                applicationContext.startService(passThroughIntent);
+                return;
+            } catch (ClassNotFoundException e) {
+                // Class not found. Try the next service
+            }
+        }
```

Comment:
is there something to be done here if there are services and none of them have been found?